README.md

```diff
@@ -1,148 +1,138 @@
-
+#### Installing Dependencies
 ```bash
-dfx deploy --network local pos-icp-integration-backend --argument '(0)'
+yarn install
 ```
 
-## Testing mainnet canister with Python
-
+### Formatting and Linting
+To keep the code clean and consistent, run the following command:
 ```bash
-# create virtual environment
-python3.11 -m venv .venv
-
-# activate virtual environment
-source .venv/bin/activate
-
-# install dependencies
-pip install -r requirements.txt
-
-# run the test
-python query_canister.py
+npx prettier --write --plugin=prettier-plugin-motoko **/*.mo
 ```
 
-source code of canister and expected responses can be reviewed at `src/pos-icp-integration-backend/main.mo`.
+### Local Project Deployment
+Follow these steps to run the project locally:
+
+1. **Start the local replica:**
+   ```bash
+   dfx start --clean --background
+   ```
+
+2. **Deploy the Internet Identity canister:**
+   ```bash
+   dfx deploy --network local internet_identity
+   ```
+
+3. **Save the current Principal:**
+   ```bash
+   export OWNER=$(dfx identity get-principal)
+   ```
+
+4. **Deploy the ckBTC ledger canister:**
+   ```bash
+   dfx deploy --network local --specified-id mxzaz-hqaaa-aaaar-qaada-cai icrc1_ledger --argument '
+   (variant {
+     Init = record {
+       token_name = "Local ckBTC";
+       token_symbol = "LCKBTC";
+       minting_account = record {
+         owner = principal "'${OWNER}'";
+       };
+       initial_balances = vec {
+         record {
+           record {
+             owner = principal "'${OWNER}'";
+           };
+           100_000_000_000;
+         };
+       };
+       metadata = vec {};
+       transfer_fee = 10;
+       archive_options = record {
+         trigger_threshold = 2000;
+         num_blocks_to_archive = 1000;
+         controller_id = principal "'${OWNER}'";
+       }
+     }
+   })
+   '
+   ```
+
+5. **Deploy the index canister:**
+   ```bash
+   dfx deploy --network local icrc1_index --argument 'opt variant { Init = record { ledger_id = principal "mxzaz-hqaaa-aaaar-qaada-cai"; } }'
+   ```
+
+6. **Deploy the pos-icp-integration-backend canister:**
+   ```bash
+   dfx deploy --network local pos-icp-integration-backend --argument '(0)'
+   ```
+
+### Testing the Mainnet Canister with Python
+
+1. **Create and activate the virtual environment:**
+   ```bash
+   python3.11 -m venv .venv
+   source .venv/bin/activate
+   ```
+
+2. **Install dependencies:**
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+3. **Run the test:**
+   ```bash
+   python query_canister.py
+   ```
+
+The canister's source code and expected responses can be reviewed in `src/pos-icp-integration-backend/main.mo`.
+
+### Future Improvements
+Planned improvements include:
+
+- **Multitoken Payment**: Support for multiple tokens such as ckBTC, ICP, ckETH, CLT, and others.
```

Comment:
Support for multiple ICRC1 and ICRC2 compliant tokens such as ckBTC, ICP, ckETH, CLT (Cafinca Loyalty Token), and others.